x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/bulk_actions/use_bulk_actions.tsx

```diff
@@ -217,11 +244,8 @@ export const useBulkActions = ({
             : { ids: selectedRuleIds },
         });
 
-        // show bulk edit confirmation window only if there is at least one failed rule
-        const hasFailedRules = (dryRunResult?.failedRulesCount ?? 0) > 0;
-
-        if (hasFailedRules && (await confirmBulkEdit()) === false) {
-          // User has cancelled edit action or there are no custom rules to proceed
+        // User has cancelled edit action or there are no custom rules to proceed
+        if ((await showBulkActionConfirmation(dryRunResult, BulkAction.edit)) === false) {
```

Comment:
Nit: same here. The comment helps, but the condition is relatively hard to read.